src/toil/batchSystems/kubernetes.py

```diff
@@ -52,8 +54,19 @@
 from toil.lib.humanize import human2bytes
 from toil.resource import Resource
 
+from toil.lib.retry import retry
+
 logger = logging.getLogger(__name__)
+     
+def exception_found(e):
+    if e is urllib3.exceptions.MaxRetryError or e is http.client.GATEWAY_TIMEOUT or e is kubernetes.client.rest.ApiException:
+        return False
+    return True
 
+def retry_kubernetes(t=5, retry_for=300, retry_while=exception_found):
+    return retry(delays=(t, 1*t, 1*t, 4*t, 16*t, 64*t),
```

Comment:
We shouldn't hardcode all these numbers.

Since `retry_kubernetes` isn't called with non-default timeouts, we could just drop all the timeout numbers and parameters and call `retry` with only `predicate=` passed.

Or, we could factor out the defaults from [here](https://github.com/DataBiosphere/toil/blob/e6dea9602ca05dcf28b190be97673fd57f69022f/src/toil/jobStores/aws/utils.py#L344-L345) into `toil.lib.retry` and pull them in here. There's no reason I can see for Kubernetes retries to have different timing characterisitcs than other retries Toil does.